packages/@guardian/source-foundations/src/utils/focus-style-manager.ts

```diff
@@ -0,0 +1,85 @@
+// FYI
+// src/core/foundations/src/utils/focus-style-manager.ts SYMLINKS TO
+// packages/@guardian/source-foundations/src/utils/focus-style-manager.ts
+
+const TAB_KEY_CODE = 9;
+
+/**
+ * Source: Blueprint https://blueprintjs.com/
+ * https://github.com/palantir/blueprint/blob/06a186c90758bbdca604ed6d7bf639c3d05b1fa0/packages/core/src/common/interactionMode.ts
+ * A nifty little class that maintains event handlers to add a class to the container element
+ * when entering "mouse mode" (on a `mousedown` event) and remove it when entering "keyboard mode"
+ * (on a `tab` key `keydown` event).
+ * Requires @babel/plugin-proposal-class-properties
+ *
+ * Modifications to original source:
+ * - Add return types to `isActive`, `start` and `stop` functions
+ */
+export class InteractionModeEngine {
+	private isRunning = false;
+
+	// tslint:disable-next-line:no-constructor-vars
+	constructor(private container: Element, private className: string) {}
+
+	/** Returns whether the engine is currently running. */
+	public isActive(): boolean {
+		return this.isRunning;
+	}
+
+	/** Enable behavior which applies the given className when in mouse mode. */
+	public start(): void {
+		this.container.addEventListener('mousedown', this.handleMouseDown);
+		this.isRunning = true;
+	}
+
+	/** Disable interaction mode behavior and remove className from container. */
+	public stop(): void {
+		this.reset();
+		this.isRunning = false;
+	}
```

Comment:
I've modified this to fix some eslint errors but not sure if it's better to leave as-is?

Reply:
i think it's good to have it here – it's our at this point, even if derived from blueprint (none of their changes will show up here unless we make them)